packages/flutter/test/material/input_decorator_test.dart

```diff
@@ -1863,6 +1863,43 @@ void main() {
     expect(tester.getTopRight(find.byKey(pKey)).dx, tester.getTopLeft(find.text('text')).dx);
   });
 
+  testWidgets('InputDecorator shows error widget', (WidgetTester tester) async {
+    // when error is defined
+    await tester.pumpWidget(
+      buildInputDecorator(
+        useMaterial3: useMaterial3,
+        decoration: const InputDecoration(
+          error: Text(
+            'error',
+            style: TextStyle(fontSize: 20.0),
+          ),
+        ),
+      ),
+    );
+
+    expect(tester.getSize(find.byType(InputDecorator)), const Size(800.0, 76.0));
+    expect(tester.getSize(find.text('error')).height, 20.0);
+    expect(tester.getTopLeft(find.text('error')), const Offset(0.0, 56.0));
+    //When errorText and error is defined then error will be shown
+    await tester.pumpWidget(
+      buildInputDecorator(
+        useMaterial3: useMaterial3,
+        decoration: const InputDecoration(
+          error: Text(
+            'error',
+            style: TextStyle(fontSize: 22.0),
+          ),
+          errorText: 'errorText',
+        ),
+      ),
+    );
+
+    expect(tester.getSize(find.byType(InputDecorator)), const Size(800.0, 78.0));
+    expect(tester.getSize(find.text('error')).height, 22.0);
+    expect(tester.getTopLeft(find.text('error')), const Offset(0.0, 56.0));
+    expect(find.text('errorText'), findsNothing);
+  });
+
   testWidgets('InputDecorator tall prefix with border', (WidgetTester tester) async {
```

Comment:
nit: This test is added between two tests that are closely related to each other. That makes those two tests harder to read. Instead, put it somewhere where it's not separating a group of closely-related tests.

One good place would be just before "InputDecorator prefix/suffix texts".